implementations/elixir/ockam/ockam/lib/ockam/credential/authenticator/direct/client.ex

```diff
@@ -0,0 +1,121 @@
+defmodule Ockam.Credential.Authenticator.Direct.Client.AddMemberRequest do
+  @moduledoc """
+  Request data structure for Credential Authority authenticator API add_member
+  """
+  use TypedStruct
+
+  typedstruct do
+    plugin(Ockam.TypedCBOR.Plugin)
+    field(:identity_id, String.t(), minicbor: [key: 1])
+    field(:attributes, %{String.t() => String.t()}, minicbor: [key: 2])
+  end
+end
+
+defmodule Ockam.Credential.Authenticator.Direct.Client.AttributesEntry do
+  @moduledoc """
+  Data structure representing attribute entry for Credential Authority authenticator API list_members
+  """
+  use TypedStruct
+
+  typedstruct do
+    plugin(Ockam.TypedCBOR.Plugin)
+    ## Rust encodes attribute values as a list of bytes
+    field(:attributes, %{String.t() => {:list, :integer}}, minicbor: [key: 1])
+    field(:added_at, integer(), minicbor: [key: 2])
+    field(:expires, integer() | nil, minicbor: [key: 3])
+    field(:attested_by, String.t() | nil, minicbor: [key: 4])
+  end
+end
+
+defmodule Ockam.Credential.Authenticator.Direct.Client.ListMembersResponse do
+  @moduledoc """
+  Response data structure for Credential Authority authenticator API list_members
+
+  This API returns a map of identity identifiers to attribute entries
+  """
+
+  alias Ockam.Credential.Authenticator.Direct.Client.AttributesEntry
+
+  @schema {:map, :string, AttributesEntry.minicbor_schema()}
+
+  def decode_strict(data) do
+    with {:ok, decoded} <- Ockam.TypedCBOR.decode_strict(@schema, data) do
+      ## Convert attribute values to binaries
+
+      Enum.map(decoded, fn {id, entry} ->
+        attributes =
+          Map.get(entry, :attributes, %{})
+          |> Enum.map(fn {key, val} -> {key, :erlang.list_to_binary(val)} end)
+          |> Map.new()
+
+        {id, Map.put(entry, :attributes, attributes)}
+      end)
+      |> Map.new()
+    end
+  end
+end
+
+defmodule Ockam.Credential.Authenticator.Direct.Client do
+  @moduledoc """
+  API client for Credential Authority authenticator service
+
+  Allows to add new members to credential authority
+  """
+
+  alias Ockam.API.Client, as: ApiClient
+  alias Ockam.API.Response, as: ApiResponse
+
+  alias Ockam.Credential.Authenticator.Direct.Client.AddMemberRequest
+  alias Ockam.Credential.Authenticator.Direct.Client.AttributesEntry
+  alias Ockam.Credential.Authenticator.Direct.Client.ListMembersResponse
+
+  @spec add_member(String.t(), %{String.t() => String.t()}, Ockam.Address.route()) ::
+          :ok | {:error, any()}
+  def add_member(identity_id, attributes, api_route) do
+    request =
+      AddMemberRequest.encode!(%AddMemberRequest{
+        identity_id: identity_id,
+        attributes: attributes
+      })
+
+    case ApiClient.sync_request(:post, "/", request, api_route) do
+      {:ok, %ApiResponse{status: 200}} ->
+        :ok
```

Comment:
Does it make any sense for this branch to return the revised list of members as part of its response body? I know that's implemented elsewhere if we chose to pursue it.

Reply:
AddMember API doesn't return them and I didn't want to change the existing APIs.